### PR TITLE
adding 'pytorch_geometric'

### PR DESCRIPTION
**Pull request by steelrose880**
Please add the Pytorch Geometric package to the Kaggle kernels: https://pytorch-geometric.readthedocs.io/en/latest/

https://github.com/rusty1s/pytorch_geometric

This is one of the few Graph Neural Network libraries in pytorch

### CODE REVIEW

**Comment by steelrose880** [comment]
please don't merge yet !!! need to add the other packages

File: tests/test_pytorch_geometric.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    import torch_sparse
+    import torch_scatter
+    from torch_geometric.data import Data
+    from torch_sparse import coalesce
+    import torch_geometric
+    from torch_geometric.nn import SplineConv
+
+class TestTorchGeometric(unittest.TestCase):
+    def test_scatter(self):
+        if torch.cuda.is_available():
```
Comment (rosbo) [inline tests/test_pytorch_geometric.py]: Why do you want to skip this test if a GPU is available?

File: tests/test_pytorch_geometric.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        x = torch.randn((num_nodes, in_channels))
+        pseudo = torch.rand((edge_index.size(1), 3))
+
+        conv = SplineConv(in_channels, out_channels, dim=3, kernel_size=5)
+        assert conv.__repr__() == 'SplineConv(16, 32)'
+        with torch_geometric.debug():
+           assert conv(x, edge_index, pseudo).size() == (num_nodes, out_channels)
+
+    def test_geometric(self):
+        if torch.cuda.is_available():
```
Comment (rosbo) [inline tests/test_pytorch_geometric.py]: Same question here

File: tests/test_pytorch_geometric.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        x = torch.tensor([[1, 3, 5], [2, 4, 6]], dtype=torch.float).t()
+        edge_index = torch.tensor([[0, 0, 1, 1, 2], [1, 1, 0, 2, 1]])
+        data = Data(x=x, edge_index=edge_index).to(torch.device('cpu'))
+
+        N = data.num_nodes
+
+        self.assertTrue(data.x.tolist() == x.tolist())
+        self.assertTrue(data['x'].tolist() == x.tolist())
+
+        self.assertTrue(sorted(data.keys) == ['edge_index', 'x'])
```
Comment (rosbo) [inline tests/test_pytorch_geometric.py]: Use self.assertCountEqual(...) to compare to list when you don't care about the ordering:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

Applies throughout to all list comparison.

File: tests/test_pytorch_geometric.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+        self.assertTrue(data.__repr__() == 'Data(edge_index=[2, 4], x=[3, 2])')
+
+        dictionary = {'x': data.x, 'edge_index': data.edge_index}
+        data = Data.from_dict(dictionary)
+        self.assertTrue(sorted(data.keys) == ['edge_index', 'x'])
+
+        self.assertTrue(data.is_undirected())
+
+        self.assertTrue(data.num_nodes == 3)
```
Comment (rosbo) [inline tests/test_pytorch_geometric.py]: Use self.assertEqual to get a nicer diff (helps for debugging test failure).

**Comment by rosbo** [comment]
The CI failure didn't seem related to this PR. Please rebase your change to get the latest changes from master. (note: master is green at the moment).

**Comment by steelrose880** [comment]
should be finalized now, gpu tests are ok, cpu tests fail outside of my functionality (test_gcloud.TestGcloud in test_version)

btw I'm not sure I did the rebase correctly, now it shows too many changes incl changes which I didn't do

**Comment by rosbo** [comment]
Hard to say from the history which git command you ran and how it went wrong so I have created a new PR #675.

**Pull request closed without merging**